spec.html

```diff
@@ -12560,10 +12572,11 @@ <h1>Runtime Semantics: EvaluateCall ( _func_, _ref_, _arguments_, _tailPosition_
           1. If Type(_ref_) is Reference, then
             1. If IsPropertyReference(_ref_) is *true*, then
               1. Let _thisValue_ be GetThisValue(_ref_).
-            1. Else the base of _ref_ is an Environment Record,
+            1. Else,
+              1. Assert: the base of _ref_ is an Environment Record.
               1. Let _refEnv_ be GetBase(_ref_).
               1. Let _thisValue_ be _refEnv_.WithBaseObject().
-          1. Else Type(_ref_) is not Reference,
+          1. Else,
```

Comment:
this seems to have been lost:
```suggestion
          1. Else,
              1. Assert: ! Type(_ref_) is not Reference.
```

Reply:
This is intentional: it is `Else` clause for `If Type(_ref_) is Reference`, so asserting `Type(_ref_) is not Reference` in there seems useless.

We've removed more of those in #1413.

Reply:
aha, makes sense :-)